### PR TITLE
ARROW-4743: [Java] Add javadoc missing in classes and methods in java…

### PR DESCRIPTION
**Pull request by emkornfield**
This is the result finding errors after applying the following diff to suppressions.xml (note for some reason this only seems to highlight method level documentation issue not class ones, if anyone can point out the right configuration knob to automatically detect missing class docs as well I would appreciate it):
--- a/java/dev/checkstyle/suppressions.xml
+++ b/java/dev/checkstyle/suppressions.xml
@@ -24,17 +24,12 @@
   <suppress checks="JavadocPackage" files=".*[\\/]maven-archetypes[\\/].*"/>
   <suppress checks="JavadocPackage" files=".*[\\/]examples[\\/].*"/>
 
-  
-  <suppress checks=".*" files=".+[\\/]generated[\\/].+\.java" />
-  <suppress checks=".*" files=".+[\\/]generated-sources[\\/].+\.java" />
+  
   <suppress checks=".*" files=".+[\\/]generated-test-sources[\\/].+\.java" />
 
   
   <suppress checks="Header" files="AutoCloseables.java|Collections2.java" />
 
-  
-  <suppress checks="JavadocMethod" message="Missing a Javadoc comment." files=".*" />

### CODE REVIEW

**Comment by wesm** [comment]
@pravindra can you take a look? Anything additional you want to do here?

**Comment by emkornfield** [comment]
@pravindra do you think you will have some time to look at this? Or should i try to ping another committer familiar with the Java codebase?

File: java/memory/src/main/java/org/apache/arrow/memory/OutOfMemoryException.java [hunk 1]
```diff
@@ -17,7 +17,13 @@
 
 package org.apache.arrow.memory;
 
-
+/**
+ * Indicates memory could not be allocated for Arrow buffers.
+ *
+ * <p>This is different from {@linkplain OutOfMemoryError} which indicates the JVM
+ * is out of memory.  However, allocation of buffers is generally capped to a fixed
+ * size in application level code.
```
Comment (siddharthteotia) [inline java/memory/src/main/java/org/apache/arrow/memory/OutOfMemoryException.java]: I am not sure if the "however" part is necessary.

Comment (siddharthteotia) [reply java/memory/src/main/java/org/apache/arrow/memory/OutOfMemoryException.java]: We can probably mention that this errors indicates that we have run out of direct memory (which is a static limit) on the allocator when it is created.

File: java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java [hunk 2]
```diff
@@ -31,6 +31,12 @@
 import io.netty.buffer.UnsafeDirectLittleEndian;
 import io.netty.util.internal.OutOfDirectMemoryError;
 
+/**
+ * A base-class that that implements all functionality of {@plainlink BufferAllocator}s.
```
Comment (siddharthteotia) [inline java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java]: typo

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java [hunk 3]
```diff
@@ -928,6 +956,10 @@ public ArrowBuf setBytes(int index, ByteBuf src, int srcIndex, int length) {
     return this;
   }
 
+  /**
+   * Copies length bytes from src starting as srcIndex
```
Comment (siddharthteotia) [inline java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: typo "at"

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java [hunk 4]
```diff
@@ -41,6 +41,17 @@
 
 import io.netty.util.internal.PlatformDependent;
 
+/**
+ * ArrowBuf is the abstraction around raw byte byte arrays that
```
Comment (siddharthteotia) [inline java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: We can probably mention that ArrowBuf is a facade over an underlying chunk of memory (UDLE). Essentially ArrowBuf provides APIs to safely interact with underlying memory UDLE and abstracts away the concept of memory address etc.

File: java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java [hunk 5]
```diff
@@ -17,6 +17,12 @@
 
 package org.apache.arrow.memory;
 
+/**
+ * Configuration class to determine if bounds checking should be turned on or off.
+ *
+ * <p>Bounds checking is turned on by setting the system property
```
Comment (praveenbingo) [inline java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java]: this looks in conflict with the code.

Setting the property seems to turn off bounds checking, while setting to false(which is the default) turns it on.

**Review by praveenbingo — CHANGES_REQUESTED** [review]

**Comment by pravindra** [comment]
sorry, I missed this thread. thanks @siddharthteotia and @praveenbingo for looking into this.

File: java/memory/src/main/java/org/apache/arrow/memory/OutOfMemoryException.java (hunk 1 above)
Comment (emkornfield) [reply java/memory/src/main/java/org/apache/arrow/memory/OutOfMemoryException.java]: reworded.

File: java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java (hunk 5 above)
Comment (emkornfield) [reply java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java]: Good catch, don't know how I missed that.  reworded.  If I read it correctly now it seems like both system properties need to be set to true to turn off bounds checking, this seems like odd behavior (requiring a deprecated parameter)?

**Comment by emkornfield** [comment]
@siddharthteotia and @praveenbingo thanks for the review, I tried to address your comments.

**Review by siddharthteotia — APPROVED** [review]
LGTM. Thanks @emkornfield

File: java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java (hunk 5 above)
Comment (praveenbingo) [reply java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java]: definitely odd :) the drill parameter should be marked deprecated (and have a default?)

**Review by praveenbingo — APPROVED** [review]
LGTM.

File: java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java (hunk 5 above)
Comment (emkornfield) [reply java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java]: ARROW-4851

**Comment by praveenbingo** [comment]
@pravindra @siddharthteotia - can one of you please help merge this.

**Pull request closed without merging**